cpp/src/arrow/util/uri.h

```diff
@@ -45,6 +45,9 @@ class ARROW_EXPORT Uri {
   /// explicit scheme.
   std::string scheme() const;
 
+  /// Convenience function that returns true if the scheme() is "file"
+  bool is_file_scheme() const;
```

Comment:
yeah, I thought it was interesting that there is an `is_file_uri_` property, but since there is a `port()` function that returns the `port_` property, I figured I would just keep consistent.